drake/systems/plants/RigidBodyManipulator.h

```diff
@@ -356,68 +246,65 @@ class DLLEXPORT_RBM RigidBodyManipulator
 
   template <typename Scalar>
-  GradientVar<Scalar, Eigen::Dynamic, 1> inverseDynamics(KinematicsCache<Scalar>& cache, const eigen_aligned_unordered_map<RigidBody const *, GradientVar<Scalar, TWIST_SIZE, 1> >& f_ext, const GradientVar<Scalar, Eigen::Dynamic, 1>& vd, int gradient_order = 0) const;
+  Eigen::Matrix<Scalar, Eigen::Dynamic, 1> inverseDynamics(KinematicsCache<Scalar>& cache, const eigen_aligned_unordered_map<RigidBody const *, Eigen::Matrix<Scalar, TWIST_SIZE, 1> >& f_ext, const Eigen::Matrix<Scalar, Eigen::Dynamic, 1>& vd) const;
 
   template <typename DerivedV>
-  GradientVar<typename DerivedV::Scalar, Eigen::Dynamic, 1> frictionTorques(Eigen::MatrixBase<DerivedV> const & v, int gradient_order = 0) const;
+  Eigen::Matrix<typename DerivedV::Scalar, Eigen::Dynamic, 1> frictionTorques(Eigen::MatrixBase<DerivedV> const & v) const;
 
-  template <typename DerivedPoints>
-  Eigen::Matrix<typename DerivedPoints::Scalar, Eigen::Dynamic, DerivedPoints::ColsAtCompileTime> forwardKin(
-      const KinematicsCache<typename DerivedPoints::Scalar> &cache, const Eigen::MatrixBase<DerivedPoints> &points, int current_body_or_frame_ind, int new_body_or_frame_ind, int rotation_type) const
+  template <typename Scalar, typename DerivedPoints> // not necessarily any relation between the two; a major use case is having an AutoDiff KinematicsCache, but double points matrix
+  Eigen::Matrix<Scalar, Eigen::Dynamic, DerivedPoints::ColsAtCompileTime> forwardKin(
+      const KinematicsCache<Scalar> &cache, const Eigen::MatrixBase<DerivedPoints> &points, int current_body_or_frame_ind, int new_body_or_frame_ind, int rotation_type) const
   {
-    cache.checkCachedKinematicsSettings(false, false, false, "forwardKin"); // rely on forwardJac for gradient cache check
+    cache.checkCachedKinematicsSettings(false, false, "forwardKin"); // rely on forwardJac for gradient cache check
 
     int npoints = static_cast<int>(points.cols());
-    typedef typename DerivedPoints::Scalar Scalar;
 
     // compute rotation and translation
-    auto T = relativeTransform(cache, new_body_or_frame_ind, current_body_or_frame_ind, 0).value();
-    auto R = T.template topLeftCorner<SPACE_DIMENSION, SPACE_DIMENSION>();
-    auto p = T.template topRightCorner<SPACE_DIMENSION, 1>();
+    auto T = relativeTransform(cache, new_body_or_frame_ind, current_body_or_frame_ind);
 
     // transform points to new frame
     Eigen::Matrix<Scalar, Eigen::Dynamic, DerivedPoints::ColsAtCompileTime> x(SPACE_DIMENSION + rotationRepresentationSize(rotation_type), npoints);
-    x.template topRows<SPACE_DIMENSION>().colwise() = p;
-    x.template topRows<SPACE_DIMENSION>().noalias() += R * points;
+    x.template topRows<SPACE_DIMENSION>().noalias() = T * points.template cast<Scalar>();
 
     // convert rotation representation
-    auto qrot = rotmat2Representation(R, rotation_type);
+    auto qrot = rotmat2Representation(T.linear(), rotation_type);
```

Comment:
This particular one was to clean up the signature of `relativeTransform`. It should have always returned a Transform, but we needed gradients, so we needed it to return a GradientVar corresponding to a 4x4 matrix instead. The method signature of `relativeTransform` was going to change anyway to get rid of GradientVar, so I took the opportunity to do it right.

Believe it or not, I do really try to minimize the number of changes in these pull requests. This one just had a large number of method signature changes that couldn't really be done gradually, and require all method calls to be adapted.